interfaces/mount/spec.go

```diff
@@ -245,11 +269,16 @@ func unclashMountEntries(entries []osutil.MountEntry) []osutil.MountEntry {
 		FirstIndex int
 		// Number of entries having this same mount point
 		Count int
-		// Merged options for the entries on this mount point
-		Options []string
 	}
 	entriesByMountPoint := make(map[string]*clashingEntry, len(entries))
+	var ensureDirEntries []osutil.MountEntry
 	for i := range entries {
+		// Gather ensure-dir mounts, do not unclash
```

Comment:
The proposed strategy is perhaps overly simplistic.

Reply:
Yes, that is my conclusion as well, and because ensuring directories only act on missing directories this seems inherently safe.

The current unclash and merge logic is not mount kind aware, which makes sense for real mounts that can be merged or renamed. 

It is possible to merge ensure-dir entries, but given that the new mkdirAllWithin() implementation created for ensure-dir is relatively cheap when for existing parent directories, I propose to keep things simple.